lib/valkyrie/storage/memory.rb

```diff
@@ -10,18 +10,21 @@ def initialize
     # @param original_filename [String]
     # @param resource [Valkyrie::Resource]
     # @return [Valkyrie::StorageAdapter::StreamFile]
-    def upload(file:, original_filename:, resource: nil)
-      identifier = Valkyrie::ID.new("memory://#{resource.id}")
-      cache[identifier] = Valkyrie::StorageAdapter::StreamFile.new(id: identifier, io: file)
+    def upload(file:, original_filename:, resource:)
+      key = next_version_key(resource: resource)
+      put(key, Valkyrie::StorageAdapter::StreamFile.new(id: to_id(key), io: file))
```

Comment:
This is a non-blocking comment, but I'm wondering about having `Valkyrie::StorageAdapter::StreamFile` being responsible for casting the input to an id. Is that feasible? It seems like it would reduce the chance for improper passing of IDs.

Also thinking of leveraging something like PowerConverter for this as well.